### PR TITLE
Part of #11964 - Move RideProperties and rideBonusValue to RideTypeDescriptors

### LINKED ISSUE #11964: Move remaining fields/arrays to RideTypeDescriptor
- [x] RideProperties - https://github.com/OpenRCT2/OpenRCT2/pull/11967
- [x] rideBonusValue - https://github.com/OpenRCT2/OpenRCT2/pull/11967
- [x] RideRatings - https://github.com/OpenRCT2/OpenRCT2/pull/11975
- [x] RideRatingsCalculateFuncTable - https://github.com/OpenRCT2/OpenRCT2/pull/11983
- [x] RideColourPresets - https://github.com/OpenRCT2/OpenRCT2/pull/11987
- [x] RideColourKey - https://github.com/OpenRCT2/OpenRCT2/pull/11996
- [x] Covered track piece availability - https://github.com/OpenRCT2/OpenRCT2/pull/11981
- [x] RideTypeEnumNames - https://github.com/OpenRCT2/OpenRCT2/pull/11999

Additionally:
- [x] Move RideTypeTrackPaintFunctionsOld to TestPaint.

### PR DESCRIPTION
**Pull request by frutiemax**
Merge RideProperties and RideBonusValue tables to RideTypeDescriptors table.

### CODE REVIEW

File: src/openrct2-ui/windows/Ride.cpp [hunk 1]
```diff
@@ -3321,8 +3321,9 @@ static void window_ride_mode_tweak_increase(rct_window* w)
     if (ride == nullptr)
         return;
 
-    uint8_t maxValue = RideProperties[ride->type].max_value;
-    uint8_t minValue = gCheatsFastLiftHill ? 0 : RideProperties[ride->type].min_value;
+    const rct_ride_properties& rideProperties = RideTypeDescriptors[ride->type].Properties;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp:3324]:
```suggestion
    const auto& rideProperties = RideTypeDescriptors[ride->type].Properties;
```

File: src/openrct2/ride/RideData.h [hunk 2]
```diff
@@ -126,6 +126,9 @@ struct RideTypeDescriptor
     /** rct2: 0x0097D7CB */
     uint8_t PhotoItem;
 
+    rct_ride_properties Properties;
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideData.h:129]: Rename this field to `OperationSettings` and move it to below `DefaultMode`. Also add `/** rct2: 0x0097CF40 */` above it,

File: src/openrct2/ride/RideData.h [hunk 3]
```diff
@@ -295,6 +296,8 @@ constexpr const RideTypeDescriptor DummyRTD =
     SET_FIELD(DefaultPrices, { 20, 20 }),
     SET_FIELD(DefaultMusic, MUSIC_STYLE_GENTLE),
     SET_FIELD(PhotoItem, SHOP_ITEM_PHOTO),
+    SET_FIELD(Properties, {0,0,0,0,0,0}),
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideData.h:299]: Format all of these in the same way other structs are, with spaces after { and before } and with one space after each comma. E.g.:

```suggestion
    SET_FIELD(Properties, { 0, 0, 0, 0, 0, 0 }),
```

**Comment by Gymnasiast** [comment]
Let's keep this PR limited to RideProperties and rideBonusValue for now. I want to avoid making this unreviewable.

**Comment by Gymnasiast** [comment]
Can you also (in a separate commit) rename the `rct_ride_properties` struct to `RideOperatingSettings` and rename all its fields to use TitleCase?

File: src/openrct2/ride/RideData.h (hunk 2 above)
Comment (frutiemax) [reply src/openrct2/ride/RideData.h:129]: There are strange compilation errors when I move OperatingSettings below DefaultMode.

Comment (frutiemax) [reply src/openrct2/ride/RideData.h:129]: `1>C:\Users\lucas\OneDrive\Documents\openrct2-frutiemax\OpenRCT2\src\openrct2\ride\RideData.h(293,5): error C2440: 'initializing': cannot convert from 'int' to 'RIDE_COMPONENT_TYPE' (compiling source file Context.cpp)
1>C:\Users\lucas\OneDrive\Documents\openrct2-frutiemax\OpenRCT2\src\openrct2\ride\RideData.h(293,5): message : Conversion to enumeration type requires an explicit cast (static_cast, C-style cast or function-style cast) (compiling source file Context.cpp)
1>C:\Users\lucas\OneDrive\Documents\openrct2-frutiemax\OpenRCT2\src\openrct2\ride\RideData.h(282,1): error C2131: expression did not evaluate to a constant (compiling source file Context.cpp)
1>C:\Users\lucas\OneDrive\Documents\openrct2-frutiemax\OpenRCT2\src\openrct2\ride\RideData.h(282,1): message : a non-constant (sub-)expression was encountered (compiling source file Context.cpp)`

Comment (Gymnasiast) [reply src/openrct2/ride/RideData.h:129]: You will need to change the order in all the RTDs too. This is an unfortunate limitation of MSVC.

Comment (frutiemax) [reply src/openrct2/ride/RideData.h:129]: I think the order in which SET_FIELD is called has importance. I'll modify that.

Comment (Gymnasiast) [reply src/openrct2/ride/RideData.h:129]: Indeed, that's what I meant.

**Comment by frutiemax** [comment]
There are now a bunch of conflicts...

File: src/openrct2-ui/windows/Ride.cpp [hunk 4]
```diff
@@ -3349,8 +3349,9 @@ static void window_ride_mode_tweak_increase(rct_window* w)
     if (ride == nullptr)
         return;
 
-    uint8_t maxValue = RideProperties[ride->type].max_value;
-    uint8_t minValue = gCheatsFastLiftHill ? 0 : RideProperties[ride->type].min_value;
+    const auto& rideProperties = RideTypeDescriptors[ride->type].OperatingSettings;
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/Ride.cpp:3352]:
```suggestion
    const auto& operatingSettings = RideTypeDescriptors[ride->type].OperatingSettings;
```

Also applies to other occurrences. (Sorry, could have been clearer.)

File: src/openrct2/ride/RideData.h [hunk 5]
```diff
@@ -128,6 +130,7 @@ struct RideTypeDescriptor
     uint8_t DefaultMusic;
     /** rct2: 0x0097D7CB */
     uint8_t PhotoItem;
+    uint8_t BonusValue;
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideData.h:133]: Forgot to mention: add `/** rct2: 0x0097D21E */` above this.

File: src/openrct2/ride/coaster/meta/AirPoweredVerticalCoaster.h [hunk 6]
```diff
@@ -25,6 +25,7 @@ constexpr const RideTypeDescriptor AirPoweredVerticalCoasterRTD =
     SET_FIELD(Flags, RIDE_TYPE_FLAGS_TRACK_HAS_3_COLOURS | RIDE_TYPE_FLAGS_COMMON_COASTER | RIDE_TYPE_FLAGS_COMMON_COASTER_NON_ALT | RIDE_TYPE_FLAG_PEEP_CHECK_GFORCES),
     SET_FIELD(RideModes, (1ULL << RIDE_MODE_POWERED_LAUNCH_PASSTROUGH) | (1ULL << RIDE_MODE_POWERED_LAUNCH)),
     SET_FIELD(DefaultMode, RIDE_MODE_POWERED_LAUNCH_PASSTROUGH),
+    SET_FIELD(OperatingSettings, { 30, 50,  30, 40,  40, 0 }),
```
Comment (Gymnasiast) [inline src/openrct2/ride/coaster/meta/AirPoweredVerticalCoaster.h:28]: Many of these field assignment still have too many spaces after the comma. For example, this line should read:

```suggestion
    SET_FIELD(OperatingSettings, { 30, 50, 30, 40, 40, 0 }),
```

**Comment by Gymnasiast** [comment]
I have checked the ported metadata. They are all correct.

File: src/openrct2/ride/transport/meta/Monorail.h [hunk 7]
```diff
@@ -16,30 +16,32 @@
...
+  SET_FIELD(AvailableBreakdowns, (1 << BREAKDOWN_SAFETY_CUT_OUT) | (1 << BREAKDOWN_DOORS_STUCK_CLOSED) | (1 << BREAKDOWN_DOORS_STUCK_OPEN) | (1 << BREAKDOWN_VEHICLE_MALFUNCTION)),
+  SET_FIELD(Heights, { 8, 32, 8, 9, }),
+  SET_FIELD(MaxMass, 78),
+  SET_FIELD(LiftData, { SoundId::Null, 5, 5 }),
+  SET_FIELD(UpkeepCosts, { 65, 20, 0, 10, 3, 10 }),
+  SET_FIELD(BuildCosts, { 42, 5, 50, }),
+  SET_FIELD(DefaultPrices, { 10, 0 }),
+  SET_FIELD(DefaultMusic, MUSIC_STYLE_SUMMER),
+  SET_FIELD(PhotoItem, SHOP_ITEM_PHOTO),
+  SET_FIELD(BonusValue, 60)
```
Comment (Gymnasiast) [inline src/openrct2/ride/transport/meta/Monorail.h:45]: Why did you change the indentation?

Comment (frutiemax) [reply src/openrct2/ride/transport/meta/Monorail.h:45]: That find and replace just screwed me up.

File: src/openrct2/ride/transport/meta/SuspendedMonorail.h [hunk 8]
```diff
@@ -23,14 +23,15 @@ constexpr const RideTypeDescriptor SuspendedMonorailRTD =
     SET_FIELD(StartTrackPiece, TRACK_ELEM_END_STATION),
     SET_FIELD(TrackPaintFunction, get_track_paint_function_suspended_monorail),
     SET_FIELD(Flags, RIDE_TYPE_FLAGS_TRACK_HAS_3_COLOURS | RIDE_TYPE_FLAG_CAN_SYNCHRONISE_ADJACENT_STATIONS |
-                     RIDE_TYPE_FLAG_HAS_LEAVE_WHEN_ANOTHER_VEHICLE_ARRIVES_AT_STATION |
-                     RIDE_TYPE_FLAG_HAS_DATA_LOGGING | RIDE_TYPE_FLAG_HAS_LOAD_OPTIONS | RIDE_TYPE_FLAG_HAS_VEHICLE_COLOURS |
-                     RIDE_TYPE_FLAG_HAS_TRACK | RIDE_TYPE_FLAG_HAS_LARGE_CURVES | RIDE_TYPE_FLAG_SUPPORTS_MULTIPLE_TRACK_COLOUR |
-                     RIDE_TYPE_FLAG_ALLOW_MUSIC | RIDE_TYPE_FLAG_HAS_ENTRANCE_EXIT | RIDE_TYPE_FLAG_ALLOW_MORE_VEHICLES_THAN_STATION_FITS |
-                     RIDE_TYPE_FLAG_ALLOW_MULTIPLE_CIRCUITS | RIDE_TYPE_FLAG_TRANSPORT_RIDE | RIDE_TYPE_FLAG_SHOW_IN_TRACK_DESIGNER |
-                     RIDE_TYPE_FLAG_IS_SUSPENDED),
+           RIDE_TYPE_FLAG_HAS_LEAVE_WHEN_ANOTHER_VEHICLE_ARRIVES_AT_STATION |
```
Comment (Gymnasiast) [inline src/openrct2/ride/transport/meta/SuspendedMonorail.h:26]: Getting there, but many files still have this indentation change that shouldn't be there.
